deid/dicom/groups.py

```diff
@@ -55,7 +56,10 @@ def extract_values_list(dicom, actions, fields=None):
         if action["action"] == "FIELD":
             for uid, field in subset.items():
                 if field.element.value not in ["", None]:
-                    values.add(field.element.value)
+                    if type(field.element.value) is MultiValue:
+                        values.update(field.element.value)
```

Comment:
So just so I understand - when we have a MultiValue it is a dictionary?

Reply:
When the value is [MultiValue](https://pydicom.github.io/pydicom/dev/reference/generated/pydicom.multival.MultiValue.html), the value in field.element.value is a "list of items that are all the same type", not a single element that we're adding into the values set.   

In the specific scenario we encountered, this was with the OtherPatientIDs field.  In all but two studies that we've deidentified using pydicom/deid, the OtherPatientIDs contained a single patient identifier, however with the two outlier studies, the modality actually sent multiple patient identifiers within this value - the MultiValue value was a list of strings.

Reply:
Ahh gotcha, so you are essentially doing:

```python
testset= set()
testset.add('value1')
testset.update(['value2', 'value3'])
{'value1', 'value2', 'value3'}
```

Reply:
Exactly.